glusterd2/brick/glusterfsd.go

```diff
@@ -94,20 +93,16 @@ func (b *Glusterfsd) SocketFile() string {
 		return b.socketfilepath
 	}
 
-	// This looks a little convoluted but just doing what gd1 did...
-
-	// First we form a fake path to the socket file
-	// Example: /var/lib/glusterd/vols/<vol-name>/run/<host-name>-<brick-path>
+	// First we form a key
+	// Example: key = peerid + brick path with '/' replaced by -
 	brickPathWithoutSlashes := strings.Trim(strings.Replace(b.brickinfo.Path, "/", "-", -1), "-")
-	// FIXME: The brick can no longer clean this up on clean shut down
-	fakeSockFileName := fmt.Sprintf("%s-%s", b.brickinfo.PeerID.String(), brickPathWithoutSlashes)
-	volumedir := utils.GetVolumeDir(b.brickinfo.VolumeName)
-	fakeSockFilePath := path.Join(volumedir, "run", fakeSockFileName)
+	key := fmt.Sprintf("%s-%s", b.brickinfo.PeerID.String(), brickPathWithoutSlashes)
 
-	// Then xxhash of the above path shall be the name of socket file.
+	// Then xxhash of the above key shall be the name of socket file.
 	// Example: /var/run/gluster/<xxhash-hash>.socket
 	glusterdSockDir := config.GetString("rundir")
-	b.socketfilepath = fmt.Sprintf("%s/%x.socket", glusterdSockDir, xxhash.Sum64String(fakeSockFilePath))
+	b.socketfilepath = fmt.Sprintf("%s/%x.socket", glusterdSockDir, xxhash.Sum64String(key))
```

Comment:
Use `path.Join` instead of `fmt.Sprintf` when joining paths. Using `path` methods whenever creating/editing paths is preferred to doing it manually. Convert the has to string using `strconv.FormatUint(hash, 16)`.

Reply:
Done